src/client/Microsoft.Identity.Client/Instance/Discovery/InstanceDiscoveryManager.cs

```diff
@@ -126,7 +126,12 @@ internal class InstanceDiscoveryManager : IInstanceDiscoveryManager
 
             if (autoDetectRegion)
             {
-                return await _regionDiscoveryProvider.GetMetadataAsync(new Uri(authority), requestContext).ConfigureAwait(false);
+                var autoDetectedRegion =  await _regionDiscoveryProvider.GetMetadataAsync(new Uri(authority), requestContext).ConfigureAwait(false);
```

Comment:
I will suggest renaming autoDetectedRegion to something like regionalizedMetadata.